src/main/java/school/hei/haapi/endpoint/rest/mapper/FeeMapper.java

```diff
@@ -17,4 +20,25 @@ public Fee toRestFee(school.hei.haapi.model.Fee fee) {
         .creationDatetime(fee.getCreationDatetime())
         .dueDatetime(fee.getDueDatetime());
   }
+
+  public school.hei.haapi.model.Fee toDomainFee(CreateFee createFee) {
+    return school.hei.haapi.model.Fee.builder()
+        .studentId(createFee.getStudentId())
+        .type(toDomainFeeType(Objects.requireNonNull(createFee.getType())))
+        .totalAmount(createFee.getTotalAmount())
+        .comment(createFee.getComment())
+        .creationDatetime(createFee.getCreationDatetime())
+        .dueDatetime(createFee.getDueDatetime())
+        .build();
+  }
+
+  private Fee.TypeEnum toDomainFeeType(CreateFee.TypeEnum createFeeType) {
+    String feeType = createFeeType.toString();
+    if (Fee.TypeEnum.TUITION.toString().equals(feeType)) {
+      return Fee.TypeEnum.TUITION;
+    } else if (Fee.TypeEnum.HARDWARE.toString().equals(feeType)) {
+      return Fee.TypeEnum.HARDWARE;
+    }
+    throw new BadRequestException("Unexpected feeType: " + feeType);
```

Comment:
Referring to
![bad-fee-enum-type](https://user-images.githubusercontent.com/88628732/157111055-ae036275-5715-4fd3-bca2-2b6a619fcb76.png)
this case will NEVER happen as it's already thrown in the `rest.model.Fee`. Also, it (IllegalArguementException) can't be handled to ExternalError (BadRequest) using our `InternatIntoExternalHandleException`

Reply:
Discussion needed